controllers/ProposalController.php

```diff
@@ -136,6 +141,27 @@ private function checkIfProposalExists(int $id): ?\app\models\Proposal
         throw new $notFoundException();
     }
 
+    public function actionGetFile(int $id)
+    {
+        $selectedProposal = $this->checkIfProposalExists($id);
+        $this->checkIfUserCanSeeProposal($selectedProposal->submitter_id);
+        $filepath = '../uploaded-files/proposal-related-files/' . $selectedProposal->file->path;
+        $extension = explode('.', $selectedProposal->file->path)[1];
+
+        if (!file_exists($filepath)) {
+            throw new FileDoesNotExistException();
+        }
+
+        if ($extension !== 'mp4') {
+            header('Content-Type: image/'. $extension);
+        } else {
+            header('Content-Type: video/'. $extension);
+        }
```

Comment:
You should use this function : https://www.php.net/manual/fr/function.mime-content-type.php

Reply:
Yep ! I didn't heard about it